codegen/gateway.go

```diff
@@ -429,6 +429,9 @@ type EndpointSpec struct {
 	TestFixtures map[string]*EndpointTestFixture
 	// Middlewares, meta data to add middlewares,
 	Middlewares []MiddlewareSpec
+	// HeadersIntoReq, a map from endpoint request headers to
+	// client request fields.
+	HeadersIntoReq map[string]FieldMapperEntry
```

Comment:
If we change this to `HeadersPropagate`, then we can s/populate/propagate everywhere and be consistent.